packages/suite/src/views/onboarding/steps/BasicSettings/TorSection.tsx

```diff
@@ -45,14 +48,13 @@ const Tor = ({ tor }: Props) => {
             <SwitchWrapper>
                 <Switch
                     dataTest="@onboarding/tor-switch"
-                    isChecked={tor}
+                    isChecked={isTorEnabled || torStatus === TorStatus.Enabling}
+                    isDisabled={isTorLoading}
                     onChange={() => {
-                        toggleTor(!tor);
+                        toggleTor(!isTorEnabled);
                     }}
                 />
             </SwitchWrapper>
         </TorWrapper>
```

Comment:
not as part of this PR, please

@tomasklim please create an separate issue...

Reply:
It was just a note to open a discussion. If we agree, I can create an issue. 
This component was refactored which was also not part of this issue. `loading` behaves totally different than in settings. It could spare @dahaca some time.